extensions/algebra/circuit/src/fp2_chip/addsub.rs

```diff
@@ -85,11 +51,73 @@ pub fn fp2_addsub_expr(
     )
 }
 
+// Input: Fp2 * 2
+// Output: Fp2
+#[derive(Chip, ChipUsageGetter, InstructionExecutor, InsExecutorE1)]
+pub struct Fp2AddSubChip<F: PrimeField32, const BLOCKS: usize, const BLOCK_SIZE: usize>(
+    pub Fp2Chip<F, BLOCKS, BLOCK_SIZE>,
+);
+
+impl<F: PrimeField32, const BLOCKS: usize, const BLOCK_SIZE: usize>
+    Fp2AddSubChip<F, BLOCKS, BLOCK_SIZE>
+{
+    pub fn new(
```

Comment:
why is the constructor so scary...

Reply:
can you pass in the `air` to simplify the constructor, or that doesn't make sense?